src/maliput_osm/builder/road_geometry_builder.cc

```diff
@@ -50,26 +50,34 @@ std::unique_ptr<const maliput::api::RoadGeometry> RoadGeometryBuilder::operator(
       .ScaleLength(builder_configuration_.scale_length)
       .InertialToBackendFrameTranslation(builder_configuration_.inertial_to_backend_frame_translation);
 
+  maliput_sparse::builder::BranchPointBuilder bp_builder = rg_builder.StartBranchPoints();
   for (const auto& osm_segment : osm_segments) {
     maliput_sparse::builder::JunctionBuilder junction_builder = rg_builder.StartJunction();
     junction_builder.Id(maliput::api::JunctionId{osm_segment.first});
     maliput_sparse::builder::SegmentBuilder segment_builder = junction_builder.StartSegment();
     segment_builder.Id(maliput::api::SegmentId{osm_segment.first});
 
     for (const osm::Lane& osm_lane : osm_segment.second.lanes) {
+      const maliput::api::LaneId lane_id{osm_lane.id};
       segment_builder.StartLane()
-          .Id(maliput::api::LaneId{osm_lane.id})
+          .Id(lane_id)
           .HeightBounds(maliput::api::HBounds{0., 5.})
           .StartLaneGeometry()
           .LeftLineString(osm_lane.left)
           .RightLineString(osm_lane.right)
           .EndLaneGeometry()
           .EndLane();
+      // Connect predecessors.
+      for (const auto& predecessor : osm_lane.predecessors) {
+        const maliput::api::LaneId predecessor_lane_id{predecessor.first};
+        bp_builder.Connect(lane_id, maliput::api::LaneEnd::kStart, predecessor_lane_id, predecessor.second);
+      }
```

Comment:
Because of each osm_lane contains the information about both predecessors and successors it isn't necessary to iterate the successors, given that the connection for the information of the successor for this lane X will be covered by the predecessors of the lane X's successors's that has X as predecessor.

Wdyt about this @agalbachicar ? Should we change the osm lane struct to ask for only the predecessor? Or the maliput_sparse::BranchPoint builder probably could simply ignore when the connection is duplicated instead of throwing: https://github.com/maliput/maliput_sparse/blob/c4663c98604f54d478e2b390ad72f4089240340a/src/builder/builder.cc#L157

Reply:
> Is it possible to face conflicting information in the map?

It is not possible. Given that predecessors/successors are determined by having points in the boundary description that are shared among the lanelets and it isn't information that is provided by each lanelet in particular.

A: For so, probably makes sense to reduce the OSMLane structo to only contain `successors`(aka next lanes) in a way that the connections are uniquely described.

B: An alternative is to get the parsed information in two different collections of structs:
 - collection<`osm_lane`/`osm_segment`> = WIth the information of the lane (geometry + adjacency info)(No predecessors/successors)
 - collection<connection> = With the information of connections  between lanelets.
 
 @agalbachicar wdyt?

Reply:
Option A: Aye!

Reply:
Gotcha!

Reply:
I just realized that when finding the successors only there is one case that won't be matched:
When lanes are connected by their start points: <----|----> . They are both predecessors for each other.

So, I think that from the predecessor and successor information of all lanes I could just process them and uniquely obtain the connections when adding to the branch point builder. I will do that. The alternative is still going to option B